### PR TITLE
feature complete + tests

### PR DESCRIPTION
**Pull request by diasdavid**

### CODE REVIEW

**Comment by jbenet** [comment]
Awesome!!!! :)

File: src/stores/config.js [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
-      return store.read('config', function (err, content) {
-        if (err) return cb(err)
+    get: function (callback) {
+      store
+        .createReadStream('config')
+        .pipe(bl(function (err, config) {
+          if (err) {
+            return callback(err)
+          }
+          callback(null, JSON.parse(config))
```
Comment (masylum) [inline src/stores/config.js]: `JSON.parse` can raise an exception, this is why I had the try/catch

Comment (diasdavid) [inline src/stores/config.js]: good catch! :) fixed here https://github.com/ipfs/js-ipfs-repo/pull/18

**Pull request merged**